src/js/view/utils/submenu-factory.js

```diff
@@ -37,14 +37,9 @@ export const makeSubmenu = (settingsMenu, name, contentItems, icon, tooltipText)
 
 export function addCaptionsSubmenu(settingsMenu, captionsList, action, initialSelectionIndex, tooltipText) {
     const captionsContentItems = captionsList.map((track, index) => {
-        const contentItemElement = SettingsContentItem(track.id, track.label, () => {
+        const contentItemElement = SettingsContentItem(track.id, track.label, (evt) => {
             action(index);
-            settingsMenu.close();
-        });
-
-        contentItemElement.uiElement().on('enter', () => {
-            action(index);
-            settingsMenu.close(true);
+            settingsMenu.close(evt.type === 'enter');
```

Comment:
Since this is being done in multiple places, I'd suggest passing the event in and have the `close` function determine how it should react based on the event type.